AirCasting/SearchAndFollow/SearchMap/SearchMapView.swift

```diff
@@ -52,38 +70,33 @@ private extension SearchMapView {
     }
     
     var searchAgainButton: some View {
-            Button {
-                withAnimation(.easeOut(duration: 0.2)) {
-                    viewModel.redoTapped()
-                }
-            } label: {
-                Text("\(Strings.SearchMapView.redoText) \(Image(systemName: "goforward"))")
-                    .font(Fonts.boldHeading3)
-                    .lineLimit(1)
-                    .scaledToFill()
-                    .multilineTextAlignment(.leading)
-                    .padding(.horizontal, 10)
-                    .padding(.vertical, 6)
+        Button {
+            withAnimation(.easeOut(duration: 0.2)) {
+                viewModel.redoTapped()
             }
-            .foregroundColor(.white)
-            .background(Color.accentColor)
-            .cornerRadius(5)
-            .padding(-3)
-            .shadow(color: Color.shadow, radius: 9, x: 0, y: 1)
-            .disabled(!viewModel.searchAgainButton)
-            .opacity(viewModel.searchAgainButton ? 1.0 : 0.0)
+        } label: {
+            Text("\(Strings.SearchMapView.redoText) \(Image(systemName: "goforward"))")
+                .font(Fonts.boldHeading3)
+                .lineLimit(1)
+                .scaledToFill()
+        }
     }
     
     var map: some View {
-        GeometryReader { geometry in
+        GeometryReader { reader in
             ZStack(alignment: .top) {
                 SearchAndFollowMap(startingPoint: viewModel.passedLocationAddress,
-                                   showRedoButton: $viewModel.searchAgainButton,
-                                   sessions: $viewModel.sessionsList).onPositionChange(action: { geoSquare in
+                                   showSearchAgainButton: $viewModel.searchAgainButton,
+                                   sessions: $viewModel.sessionsList,
+                                   selectedPointerID: $viewModel.cardPointerID)
+                .onPositionChange(action: { geoSquare in
                     viewModel.mapPositionsChanged(geoSquare: geoSquare)
                 })
-                    .padding(.top, 50)
-                    .ignoresSafeArea(.all, edges: [.bottom])
+                .onMarkerChange(action: { pointer in
+                    viewModel.markerPositionChanged(using: pointer)
```

Comment:
I think the position of the marker didn't change, but a new marker got selected 🙂